src/test/java/games/strategy/util/VersionTest.java

```diff
@@ -57,6 +57,11 @@ public void testRead3() {
   }
 
   @Test
+  public void verifyDevVersionToString() {
+    assertEquals("1.2.3.dev", new Version("1.2.3.dev").toString());
```

Comment:
Good to see the test case, though there are a few more good ones to add to cover the other functionality.
EG:
- check that toString full will override the max integer value and convert that to 'dev' when calling `toStringFull`. I'm again not 100% on why that is necessary, but assuming it's useful that would be an interesting test case.
- assert equals and compare to can now handle null values, a quick test case that the expected output comes from null comparison would verify that updated code is working as expected
- `isCompatible` does not seem to be exercised at all. I hope/suspect it would be pretty quick to throw some basic example usages of `isCompatible` into a test to verify they would work as expected.

Reply:
Added a test for toStringFull but I don't know what you are talking about with null values? Null is not valid.

@RoiEXLab  added the isCompatible() and has added the test.

Reply:
He means that passing null to equal and compareTo should return false/-1 and not result in a Nullpointerexception

Reply:
Null value test added. This should never happen right?